App.tsx

```diff
@@ -9,13 +9,33 @@ import { MidtermTabNavigator } from "./midterm/MidtermTabNavigator";
 import { WeatherDrawerNavigator } from "./weather/WeatherDrawerNavigator";
 import { ShakeToCharge } from "./battery/ShakeToCharge";
 import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import { FinalTabNavigator } from "./final/FinalTabNavigator";
+import { useEffect } from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
 
 
 
 const MainDrawer = createDrawerNavigator();
 
 const GalleryStack = createNativeStackNavigator<StackParamList>();
+
 const GalleryStackNavigator = () => {
+
+	useEffect(() => {
+		// Function to clear AsyncStorage data
+		const clearAsyncStorage = async () => {
```

Comment:
Clearing async storage when unmounting defeats its purpose. This is really no different than not using at all.